src/libs/api/middleware.http-error-handler.ts

```diff
@@ -0,0 +1,50 @@
+import { Event } from './types';
+import { jsonSafeParse, normalizeHttpResponse } from '@middy/util';
+import middy from '@middy/core';
+
+// enhance the middy HTTP error handler by incorporating a condition to forward protocolink errors.
+// https://github.com/middyjs/middy/tree/main/packages/http-error-handler
+export default function cacheMiddleware(): middy.MiddlewareObj<Event, any, any> {
+  return {
+    onError: async (request) => {
+      if (request.response !== undefined) return;
```

Comment:
```suggestion
      if (request.response) return;
```
Wondering if we need to check null here.